### PR TITLE
Alerting: Enable Alert rule tags to override PagerDuty Severity setting

### PR DESCRIPTION
**Pull request by AndrewBurian**
**What this PR does / why we need it**: 

Uses alert tags to optionally override the `severity` in the PagerDuty event.

This gives you the ability to send events of different severity to the same service without needing a separate notification channel configured.

### CODE REVIEW

File: docs/sources/alerting/notifications.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ Integration Key | Integration key for PagerDuty.
 Severity | Level for dynamic notifications, default is `critical`
 Auto resolve incidents | Resolve incidents in PagerDuty once the alert goes back to ok
 
-**Note:** The tags `Class`, `Group`, and `Component` have special meaning in the [Pagerduty Common Event Format - PD-CEF](https://support.pagerduty.com/docs/pd-cef).  If an alert panel defines these tag keys they will be transposed to the root of the event sent to Pagerduty.  This means they will be available within the Pagerduty UI and Filtering tools.
+**Note:** The tags `Severity`, `Class`, `Group`, and `Component` have special meaning in the [Pagerduty Common Event Format - PD-CEF](https://support.pagerduty.com/docs/pd-cef).  If an alert panel defines these tag keys they will be transposed to the root of the event sent to Pagerduty.  This means they will be available within the Pagerduty UI and Filtering tools. A Severity tag set on an alert will override the global Severity set on the notification channel if it's a valid level.
```
Comment (oddlittlebird) [inline docs/sources/alerting/notifications.md]:
```suggestion
**Note:** The tags `Severity`, `Class`, `Group`, and `Component` have special meaning in the [Pagerduty Common Event Format - PD-CEF](https://support.pagerduty.com/docs/pd-cef). If an alert panel defines these tag keys, then they are transposed to the root of the event sent to Pagerduty. This means they will be available within the Pagerduty UI and Filtering tools. A Severity tag set on an alert overrides the global Severity set on the notification channel if it's a valid level.
```

**Review by oddlittlebird — DISMISSED** [review]

**Comment by AndrewBurian** [comment]
I somehow managed to remove @grafana/backend-platform from the reviewers list... My bad. Can someone re-add that?

**Review by bergquist — CHANGES_REQUESTED** [review]
Hi

How is this better than using different notification channels? From my perspective that would achive the same thing.

**Comment by AndrewBurian** [comment]
It's both cleaner, and enables better division of responsibility.

In our use case, the notification channels are configured by the SRE team, and then consumed at the dashboard level by the engineers. We currently run 38 services, times 4 severity levels, thats crossing 150 notification channels that need to be manually added, and then searched for in the alerting panel. If a service changes or needs a new integration key, that's 4 copies of the same service that need updated.

I really like the addition of alert tags, and using that control to add things to the PagerDuty event fields for Class, Group, and Component has already been useful. Even if the configuration complexity wasn't being 4x'd without it, I would still argue that it's a good use of tags to bring that control to a more granular level per-alert rather than per-channel.

File: pkg/services/alerting/notifiers/pagerduty_test.go [hunk 2]
```diff
@@ -197,6 +197,7 @@ func TestPagerdutyNotifier(t *testing.T) {
 						{Key: "group", Value: "aGroup"},
 						{Key: "class", Value: "aClass"},
 						{Key: "component", Value: "aComponent"},
+						{Key: "severity", Value: "warning"},
```
Comment (bergquist) [inline pkg/services/alerting/notifiers/pagerduty_test.go]: Could you please add a test using another severity as well?

**Review by bergquist — CHANGES_REQUESTED** [review]
I agree that this is cleaner. This is something we want to achieve at a generic level rather then just for one alert notifier. But that's another issue :)

Could you please add more tests with different severity levels so that we test that it's based on input data. Once that is done I'll merge this.

**Comment by bergquist** [comment]
Thanks! I'll merge this one the build passes :)

**Comment by bergquist** [comment]
Depends on https://github.com/grafana/grafana/pull/22909

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
Thank you for contributing!

**Comment by AndrewBurian** [comment]
@oddlittlebird FYI, I think the documentation changes here are in the grafana docs under v6.7, but the code is not actually present in the v6.7.3 release, so those are out of sync. Judging by the milestone here, I think the docs aren't meant to show up until v7

**Comment by oddlittlebird** [comment]
@bergquist, could you check on this?

**Pull request merged**